### PR TITLE
Fix getProductDetails failing to return SKU details if list not ordered as in Google's inventory

### LINKED ISSUE #18: java.util.NoSuchElementException - QueryInventoryFinishedListener
QueryInventoryFinishedListener fails for entries which are not the first element in SkuDetails sku : skuList

The following code fails because it calls next on the iterator inside the sku iterator
```
//This should be caching each entry in requestSkuIter and looping through skuList
if (!sku.getSku().equalsIgnoreCase(requestSkuIter.next()) ) { 
```

Below is the stack trace

```
08-29 10:11:47.222: E/AndroidRuntime(9018): java.util.NoSuchElementException
08-29 10:11:47.222: E/AndroidRuntime(9018): 	at java.util.ArrayList$ArrayListIterator.next(ArrayList.java:576)
08-29 10:11:47.222: E/AndroidRuntime(9018): 	at jp.wizcorp.phonegap.plugin.wizPurchase.WizPurchasePlugin$2.onQueryInventoryFinished(WizPurchasePlugin.java:523)
08-29 10:11:47.222: E/AndroidRuntime(9018): 	at com.smartmobilesoftware.util.IabHelper$2$1.run(IabHelper.java:630)
```

### LINKED ISSUE #35: Error: One or more SKU was not found in Google Inventory
Hello,

I try to use your plugin but, on Android I get these logs: http://pastebin.com/kWM7HV4V 

What's mean "One or more Sku were not found in Google Inventory"? Did I miss something?

Regards,

### PR DESCRIPTION
**Pull request by jrouault**
### Patch
```getProductDetails``` now correctly returns SKU details whatever the order they are asked in. It mirrors iOS behaviour too and returns an ```UNKNOWN_PRODUCT_ID``` error if one of the ```productId``` doesn't exist in Google's inventory.
Fixes #18 and #35.

poke @ronkorving @aogilvie
@CyrilleGuimezanes It would be nice if you could have a look at this PR as well since it fixes #35.

### CODE REVIEW

File: platforms/android/src/jp/wizcorp/phonegap/plugin/wizPurchase/WizPurchasePlugin.java [hunk 1]
```diff
@@ -577,61 +577,46 @@ public void onQueryInventoryFinished(IabResult result, Inventory inventory) {
...
+				// Iterate all skus
+				while (requestSkuIter.hasNext()) {
+					String requestedSku = requestSkuIter.next();
+					SkuDetails skuDetails = inventory.getSkuDetails(requestedSku);
+
+					if (skuDetails == null) {
+						Log.d(TAG, "Invalid product id: " + requestedSku);
+						mProductDetailCbContext.error(UNKNOWN_PRODUCT_ID);
+						mProductDetailCbContext = null;
+						return;
```
Comment (ronkorving) [inline platforms/android/src/jp/wizcorp/phonegap/plugin/wizPurchase/WizPurchasePlugin.java]: Not `continue`?

**Comment by CyrilleGuimezanes** [comment]
Hello,

As @ronkorving suggest I think its better to change "return" (L596) by "continue".

I would be nice to have a log when a valid SKU is found & on "catch" statements 

Regards,

**Comment by ronkorving** [comment]
That was my first intuition, but reading further convinced me otherwise and I deleted my comment. Perhaps @jrouault could explain the reasoning better than me though.

**Comment by jrouault** [comment]
Yep, the callback is returned early with an ```UNKNOWN_PRODUCT_ID``` error to have the same behaviour as iOS. It would have been nice indeed to log every product ID not found but I prefer to keep it simple here.
If one product ID is not found, it's a serious issue that should lead to review the entire list of IAP on both stores to ensure it will not happen with any other IAP.
I have added logs on ```catch``` statements as it's a good improvement and should simplify debugging.

**Pull request merged**